### ISSUE TITLE
Missing commits in changelog when PRs are already open at release-please PR opening

### ISSUE THREAD

**Issue by jake-rvs** [googleapis/release-please#2206]
#### Environment details

  - OS: Github Runner / Ubuntu 22.04.3
  - Node.js version: 16
  - npm version: n/a
  - `release-please` version: Github Action v3

#### Steps to reproduce
Setup
- Have three branches-- `development`, `staging`, and `main`
- Have `release-please` set up to run against the `staging` branch
- We are squash-merging PRs and using merge commits to merge branches like `development` to `staging` / `staging` to `main` / etc.

1. Dev A creates a branch off of `development` -- `fix/test-1`
2. Dev B creates a branch off of `development` -- `fix/test-2`
3. `fix/test-1` is squash-merged to `development`
4. It is release time-- `development` is merged to `staging`
5. `release-please` runs against `staging` and generates release notes
6. Development continues on the `development` branch
7. `fix/test-2` is ready and is squash-merged to `development`
8. Once QA approves the release, the release PR is merged to `staging`
9. `staging` is merged into `main` 
10. `main` is merged back into `development`
11. Create another branch off of `development` -- `fix/test-3`
12. squash-merge `fix/test-3` to `development`
13. merge `development` to `staging` and wait for `release-please` to generate a PR

That PR will not have `fix/test-2`-- it appears that any PRs that are opened _after_ we push to `staging` and the `release-please` action runs, but _before_ we merge that PR to `staging` are not picked up by `release-please` on the next release.

Any suggestions on what might be going wrong here??

**Comment by ashish-fund** [comment]
@jake-rvs: Hey, I have similar issue. Did you find any solution?

**Comment by jake-rvs** [comment]
Probably not ideal, but the solution I ended up going with was not to do the thing I mentioned in the end of my comment-- specifically, make sure that you don't open any PRs _after_ you push to your _staging_ or similar branch and _before_ you merge the `release-please` generated PRs.
